Automattic/wp-calypso
# Change shape of email stats Redux data, allowing for better merging

#### Proposed Changes

This PR changes the way email stats data flows to Redux.

| Before | After |
|-------|-------|
| <img width="700" alt="CleanShot 2023-01-05 at 10 20 34@2x" src="https://user-images.githubusercontent.com/528287/210745463-8c70939a-c6b5-4baa-8995-570f8fe5f296.png"> | <img width="738" alt="CleanShot 2023-01-05 at 10 21 55@2x" src="https://user-images.githubusercontent.com/528287/210745521-d235d329-80ea-4365-8dfd-b5c8010ca295.png"> |

#### Testing Instructions

1. Apply this PR
2. Test that both the chart & country list still work.
3. Verify that when a new Redux request is made (by using the next/prev arrows, or clicking a data), the incoming data is properly merged with the existing data.

You might want to refresh a few times until Redux cache is randomly cleared.

<img width="530" alt="CleanShot 2023-01-05 at 10 21 24@2x" src="https://user-images.githubusercontent.com/528287/210745675-2f78348d-3753-4ce7-950b-deb12215523b.png">

#### Pre-merge Checklist





Related to #71696
